src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java

```diff
@@ -0,0 +1,285 @@
+  }
+
+  private Boolean isMobileVideoPaused() {
+    return Boolean.valueOf(driver.findElement(getVideoSelector()).getAttribute("paused"));
+  }
+
+  private Boolean isDesktopVideoPaused() {
+    return Boolean.valueOf(usingImaBridge(webDriver -> {
+      final JavascriptActions js = new JavascriptActions();
+
+      return js.execute("document.querySelector('video') && document.querySelector('video').paused").toString();
+    }));
+  }
+
+  public void replay() {
+    wait.forElementVisible(replayOverlaySelector).click();
+  }
+
+  public void close() {
+    wait.forElementClickable(closeButtonSelector).click();
+  }
+
+  public void closeWithJS() {
+    clickByJS(closeButtonSelector);
+  }
+
+  private void clickByJS(By selector) {
+    new JavascriptActions(driver).execute("arguments[0].click();", driver.findElement(selector));
+  }
+
+  private void clickElementInsideAd(By selector) {
+    usingAdFrame(() -> {
+      // It need to be clicked by JS, because our templates elements covers each other
+      // and there is no way to click it by just .click()
+      clickByJS(selector);
+    });
+  }
+
+  public double getVideoHeightWhilePaused() {
+    return driver.findElement(pauseOverlaySelector).getSize().getHeight();
+  }
+
+  public double getAdSlotHeight() {
+    waitForAdToLoad();
+    return driver.findElement(By.id(slot)).getSize().getHeight();
+  }
+
+  public int getProgressBarWidth() {
+    return driver.findElement(progressBarSelector).getSize().getWidth();
+  }
+
+  public boolean hasStarted() {
+    return getCurrentTime() > 0;
+  }
+
+  public boolean isVisible() {
+    return isDesktop() ?
+           usingImaBridge(webDriver -> webDriver.findElement(By.tagName("video")).isDisplayed()) :
+           driver.findElement(getVideoSelector()).isDisplayed();
+  }
+
+  public boolean isMuted() {
+    return wait.forElementVisible(speakerSelector).getAttribute("class").contains("mute");
+  }
+
+  public boolean isUnmuted() {
+    return !isMuted();
+  }
+
+  private String getTitleDesktop() {
+    return usingVideoContext(video -> video.getAttribute("title"));
+  }
+
+  private String getTitleMobile() {
+    return driver.findElement(getVideoSelector()).getAttribute("title");
+  }
+
+  public String findTitle() {
+    return isDesktop() ? getTitleDesktop() : getTitleMobile();
+  }
+
+  public void waitForVideoToStart(final long timeout) {
+    waitFor(AutoplayVuap::isVisible, timeout);
+  }
+
+  public void waitForVideoToEnd(final long timeout) {
+    waitFor(AutoplayVuap::isOverlayNoVisible, timeout);
+  }
+
+  public void waitForAdToLoad() {
+    usingAdFrame(() -> wait.forElementPresent(By.cssSelector(AD_TNG_CLICK_AREA_2_SELECTOR)));
+  }
+
+  private void waitFor(final Predicate<AutoplayVuap> predicate, final long timeout) {
+    new FluentWait<>(this)
+        .withTimeout(timeout, TimeUnit.SECONDS)
+        .pollingEvery(1, TimeUnit.SECONDS)
+        .until(predicate);
+  }
+
+  private boolean isOverlayNoVisible() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "isOverlayNoVisible" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)